shared/src/ledger/eth_bridge/storage/vote_tracked.rs

```diff
@@ -4,29 +4,23 @@ use crate::types::hash::Hash;
 use crate::types::storage::Key;
 
 #[allow(missing_docs)]
-pub const PREFIX_KEY_SEGMENT: &str = "eth_msgs";
-
-/// Get the key prefix corresponding to where details of seen [`EthereumEvent`]s
-/// are stored
-pub fn prefix() -> Key {
-    super::prefix()
-        .push(&PREFIX_KEY_SEGMENT.to_owned())
-        .expect("should always be able to construct this key")
-}
+pub const ETH_MSGS_PREFIX_KEY_SEGMENT: &str = "eth_msgs";
 
 const BODY_KEY_SEGMENT: &str = "body";
 const SEEN_KEY_SEGMENT: &str = "seen";
 const SEEN_BY_KEY_SEGMENT: &str = "seen_by";
 const VOTING_POWER_KEY_SEGMENT: &str = "voting_power";
 
-/// Generator for the keys under which details of an [`EthereumEvent`] is stored
-pub struct Keys {
-    /// The prefix under which the details are an [`EthereumEvent`] is stored
+/// Generator for the keys under which details of a vote tracked event is stored
+pub struct Keys<T: 'static> {
```

Comment:
```suggestion
pub struct Keys<T> {
```
you don't need to add trait bounds at the struct def level, generally

Reply:
It does't compile if the `: 'static` is removed

```
error[E0310]: the parameter type `T` may not live long enough
  --> shared/src/ledger/eth_bridge/storage/vote_tracked.rs:18:15
   |
15 | pub struct Keys<T> {
   |                 - help: consider adding an explicit lifetime bound...: `T: 'static`
...
18 |     _phantom: std::marker::PhantomData<&'static T>,
   |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ ...so that the reference type `&'static T` does not outlive the data it points at

For more information about this error, try `rustc --explain E0310`.
error: could not compile `namada` due to previous error
```

Reply:
ah. well that's kinda odd. alright!

Reply:
If you remove the `'static` on both is works fine.